risc0/zkvm/src/host/api/client.rs

```diff
@@ -87,8 +93,8 @@ impl Client {
         }
 
         let receipt_bytes = asset.as_bytes()?;
-        let receipt = bincode::deserialize(&receipt_bytes)?;
-        Ok(receipt)
+        let receipt_pb = pb::core::Receipt::decode(receipt_bytes)?;
+        receipt_pb.try_into()
     }
 
     /// TODO
```

Comment:
Not something that needs to be solved within this PR, but I want to note that this architecture is not yet fully documented and we'll need to finish its docs before the next release.

Reply:
Tracking in #929

Reply:
I actually think client/server is largely transparent to end users. We do need to describe the public bits and how to install the server, but I don't think we need to (or necessarily should) document the internal part since it's undergoing changes and won't be relevant (and in fact confusing) to the majority of users.

Reply:
:+1: makes sense. I do think that anything that shows up on docs.rs shouldn't have docs that just say "TODO". (Minimally something like "Internal component")

Reply:
If you see a TODO, could you mark it or link to it? I probably missed those

Reply:
The line this comment is on is one of them. I think they were introduced in a different client/server PR -- I noticed them when reviewing by reading through the generated docs, not on the PR proper. Also lines 66, 100, 137, 177, and 215 of this file, and lines 178, 183, and 189 of https://github.com/risc0/risc0/blob/main/risc0/zkvm/src/host/api/server.rs (as of commit `7a1581f160729b544025afaca525ee3a39394b6e`).